include/xtensor/xset_operation.hpp

```diff
@@ -0,0 +1,101 @@
+/***************************************************************************
+* Copyright (c) Johan Mabille, Sylvain Corlay and Wolf Vollprecht          *
+* Copyright (c) QuantStack                                                 *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#ifndef XTENSOR_XSET_OPERATION_HPP
+#define XTENSOR_XSET_OPERATION_HPP
+
+#include <algorithm>
+#include <functional>
+#include <type_traits>
+
+#include <xtl/xsequence.hpp>
+
+#include "xfunction.hpp"
+#include "xscalar.hpp"
+#include "xstrides.hpp"
+#include "xstrided_view.hpp"
+#include "xmath.hpp"
+
+namespace xt
+{
+
+    /**
+    * @ingroup logical_operators
+    * @brief isin
+    *
+    * Returns a boolean array of the same shape as element that is ``true`` where an element of
+    * ``element`` is in ``test_elements`` and ``False`` otherwise.
+    * @param element an \ref xexpression
+    * @param test_elements an array
+    * @return a boolean array
+    */
+    template <class E, class T>
+    inline auto isin(E&& element, std::initializer_list<T> test_elements) noexcept
+    {
+        auto lambda = [test_elements](const auto& t) {
+            return std::find(test_elements.begin(), test_elements.end(), t) != test_elements.end(); };
+        return make_lambda_xfunction(std::move(lambda), std::forward<E>(element));
+    }
+
+    /**
+    * @ingroup logical_operators
+    * @brief isin
+    *
+    * Returns a boolean array of the same shape as element that is ``true`` where an element of
+    * ``element`` is in ``test_elements`` and ``False`` otherwise.
+    * @param element an \ref xexpression
+    * @param test_elements an array
+    * @return a boolean array
+    */
+    template <class E, class F>
+    inline auto isin(E&& element, F&& test_elements) noexcept
+    {
+        auto lambda = [&test_elements](const auto& t) {
```

Comment:
> You may want to optimize the capture here, to avoid copies when values is a lvalue-reference. A solution for that is to introduce an intermediate structure specialized on F lvalueness and returning the lambda with capture by value or by reference.

Is this not enough?

Reply:
if test elements is a temporary, capturing by reference will lead to a dangling reference in the lambda.

Reply:
OK. Then I will work a bit harder ;)